import scala.concurrent.Future
import scala.util.{Failure, Success}
import scala.util.Random
import scala.concurrent.ExecutionContext.Implicits.global


object Tmp extends App {

  val f = Future {
    val r = Random.nextInt(500)
    if (r < 499)
      throw new RuntimeException(""bad value"")
    r
  }

  f onComplete  {
    case Success(value) => value
    case Failure(e) => e.printStackTrace
  }

  // do the rest of your work
  println(""A ...""); Thread.sleep(100)
  println(""B ...""); Thread.sleep(100)
  println(""C ...""); Thread.sleep(100)
  println(""D ...""); Thread.sleep(100)
  println(""E ...""); Thread.sleep(100)
  println(""F ...""); Thread.sleep(100)
  Thread.sleep(1000)
}
